packages/devtools_app/lib/src/screens/vm_developer/isolate_statistics_view_controller.dart

```diff
@@ -51,42 +50,42 @@ class IsolateStatisticsViewController extends DisposableController
       UnmodifiableListView(_serviceExtensions);
   List<String> _serviceExtensions = [];
 
-  Future<void> refresh() async => await switchToIsolate(isolate);
+  Future<void> refresh() async => await switchToIsolate(isolate!);
 
   Future<void> switchToIsolate(IsolateRef isolateRef) async {
     _refreshing.value = true;
 
+    final isolateId = isolateRef.id!;
     // Retrieve updated isolate information and refresh the page.
-    _isolate = await _service.getIsolate(isolateRef.id);
+    _isolate = await _service.getIsolate(isolateId);
+    final isolate = _isolate!;
     _updateTagCounters(isolate);
-    _ports = (await _service.getPorts(_isolate.id)).ports;
+    _ports = (await _service.getPorts(isolateId)).ports!;
```

Comment:
just want to verify isolateRef.id == isolate.id? before this change we were getting the id from the Isolate, and this changes behavior to grab the id from the isolateRef

Reply:
Yes, isolate IDs are stable.